spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,13 +131,48 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		CountDownLatch countDownLatch = new CountDownLatch(1);
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.debug(event);
+				countDownLatch.countDown();
+			}
+		});
+		File file = new File(testDir, "foo");
+		file.delete();
+		adapter.setFile(file);
+		QueueChannel outputChannel = new QueueChannel();
+		adapter.setOutputChannel(outputChannel);
+		adapter.setIdleEventInterval(100);
+		adapter.afterPropertiesSet();
+		adapter.start();
+		boolean noEvent = countDownLatch.await(500, TimeUnit.MILLISECONDS);
```

Comment:
Yeah, what you say sounds reasonable.
Let's take a look to variant of spying `File` and mock the call to its `exists()` with one more latch and verify that we don't have idle event afterwards, just only that one about file doesn't exist.